doc/examples/edges/plot_skeleton.py

```diff
@@ -6,59 +6,144 @@
 Skeletonization reduces binary objects to 1 pixel wide representations. This
 can be useful for feature extraction, and/or representing an object's topology.
 
-The algorithm works by making successive passes of the image. On each pass,
+``skeletonize`` works by making successive passes of the image. On each pass,
 border pixels are identified and removed on the condition that they do not
 break the connectivity of the corresponding object.
-
-This module provides an example of calling the routine and displaying the
-results. The input is a 2D ndarray, with either boolean or integer elements.
-In the case of boolean, 'True' indicates foreground, and for integer arrays,
-the foreground is 1's.
 """
 from skimage.morphology import skeletonize
-from skimage import draw
+from skimage import data
 import numpy as np
 import matplotlib.pyplot as plt
+from skimage.util import invert
 
-# an empty image
-image = np.zeros((400, 400))
-
-# foreground object 1
-image[10:-10, 10:100] = 1
-image[-100:-10, 10:-10] = 1
-image[10:-10, -100:-10] = 1
-
-# foreground object 2
-rs, cs = draw.line(250, 150, 10, 280)
-for i in range(10):
-    image[rs + i, cs] = 1
-rs, cs = draw.line(10, 150, 250, 280)
-for i in range(20):
-    image[rs + i, cs] = 1
-
-# foreground object 3
-ir, ic = np.indices(image.shape)
-circle1 = (ic - 135)**2 + (ir - 150)**2 < 30**2
-circle2 = (ic - 135)**2 + (ir - 150)**2 < 20**2
-image[circle1] = 1
-image[circle2] = 0
+# Invert the horse image
+image = invert(data.horse())
 
 # perform skeletonization
 skeleton = skeletonize(image)
```

Comment:
Wait, does this work with `horse`? It is a 3D ndarray and should cause an error.